### PR TITLE
Admin status module updates

### PR DESCRIPTION
**Pull request by C-Lodder**
Pull Request for Issue #10910 
#### Summary of Changes
- Removes "envelope" icon
- Adds "Message(s)" label after the count badge
- Adds separators in between each button group
#### Testing Instructions

Very simple. Apply the patch and match again the screenshots below:

**Before:**
before

**After:**
after

### CODE REVIEW

**Comment by MATsxm** [comment]
I have tested this item :white_check_mark: successfully on d5320ba688c8600ce3204610f5e688baaa0cf0b4<br /><br />:+1: 

I love this way. Thanks
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10918">issues.joomla.org/joomla-cms/10918</a>.</sub>

**Comment by MATsxm** [comment]
Side note, see with Multilingual Sites...
Is there a way for a "conditional" separator?
6439163c9b57aa275817d293a41e3691

**Comment by MATsxm** [comment]
Side note, see with Multilingual Sites...
Is there a way for a "conditional" separator?
89c104d074f864ad3b6624fa5a3c5439

**Comment by C-Lodder** [comment]
@MATsxm - done ;)

**Comment by andrepereiradasilva** [comment]
I have tested this item :red_circle: unsuccessfully on 73aff5a698ce3ca67cf8f4cd7bbd8791f9fed0b8<br /><br />doesn't work on RTL languages (Arabic for instance).

also it seems generatecss.php has not run to generate the css from less files.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10918">issues.joomla.org/joomla-cms/10918</a>.</sub>

File: administrator/modules/mod_status/tmpl/default.php [hunk 1]
```diff
@@ -64,10 +66,10 @@
 	$output[] = '<div class="btn-group hasTooltip ' . $inboxClass . '"'
 		. ' title="' . JText::plural('MOD_STATUS_MESSAGES', $unread) . '">'
 		. ($hideLinks ? '' : '<a href="' . $inboxLink . '">')
-		. '<span class="icon-envelope"></span>'
-		. '<span class="badge' . $active . '">' . $unread . '</span>'
+		. '<span class="badge' . $active . '">' . $unread . '</span> '
+		. JText::plural('MOD_STATUS_MESSAGES_LABEL', $unread)
```
Comment (brianteeman) [inline administrator/modules/mod_status/tmpl/default.php]: Can you look at removing the space please so that the link is not " messages" but "messages"

**Comment by brianteeman** [comment]
After applying the PR my site doesnt look exactly the same as your screenshot. The separator is not evenly spaced

File: administrator/modules/mod_status/tmpl/default.php (hunk 1 above)
Comment (brianteeman) [inline administrator/modules/mod_status/tmpl/default.php]: You will see the logout link is ok

**Comment by MATsxm** [comment]
I have tested this item :white_check_mark: successfully on 73aff5a698ce3ca67cf8f4cd7bbd8791f9fed0b8<br /><br />Works fine here with the multilingual mod_ enabled.

Didn't test on RTL<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10918">issues.joomla.org/joomla-cms/10918</a>.</sub>

**Comment by brianteeman** [comment]
When you only have multilingual mod enabled you have a seperator hanging on
its own

On 23 June 2016 at 17:21, Marc-Antoine Thevenet <notifications@github.com>
wrote:

> I have tested this item ✅ successfully on 73aff5a
> <https://github.com/joomla/joomla-cms/commit/73aff5a698ce3ca67cf8f4cd7bbd8791f9fed0b8>
>
> Works fine here with the multilingual mod_ enabled.
>
> Didn't test on RTL
> ------------------------------
> This comment was created with the J!Tracker Application
> <https://github.com/joomla/jissues> at issues.joomla.org/joomla-cms/10918
> <https://issues.joomla.org/tracker/joomla-cms/10918>.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/10918#issuecomment-228103958>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABPH8YbIJMts9eGDNSuxuRHY9O3dZQZDks5qOrKdgaJpZM4I87vN>
> .
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by ggppdk** [comment]
> When you only have multilingual mod enabled you have a seperator hanging on its own

yes because it is a different module maybe better remove it 

About RTL it works, if you copy new CSS into the rtl-css / (or into the less files),

About extra spacing in seperators it is because:
```
.btn-group + .btn-group {
    margin-left: 5px;
}
```

And in RTL same effect with seperators:
```
 .btn-group + .btn-group {
    margin-right: 5px;
    margin-left: 0px;
}
```

we could add this to fix it:
```
#status .btn-group + .btn-group {
    margin-right: 5px;
    margin-left: 0px;
}
```

----

----

But it seems like and ugly hack to do this, because the seperator is inside the btn-group, i have tested moving the separator out of the btn-group: 

```
// Output the items.
/*foreach ($output as $item)
{
	echo $item;
}*/

echo implode('<span class="btn-group separator"></span>', $output);
```

and the result is proper in both cases (RTL, non-RTL)

**Comment by infograf768** [comment]
Confused...
Could you explain the reason of the new strings:
```
+MOD_STATUS_MESSAGES_LABEL_0="Messages"
+MOD_STATUS_MESSAGES_LABEL_1="Message"
+MOD_STATUS_MESSAGES_LABEL_MORE="Messages"
```
I replaced them in the module by the existing one:
`. JText::plural('MOD_STATUS_MESSAGES', $unread)`

for which we have already the strings
```
MOD_STATUS_MESSAGES_0="%d Messages"
MOD_STATUS_MESSAGES_1="%d Message"
MOD_STATUS_MESSAGES_MORE="%d Messages"
```
and I could not see any difference in the display.

BTW I never understood why we add a tooltip for messages apparently reproducing exactly the same text as the display.

I suggest to change that part to a simple:
```
//  Print the inbox message.
if ($params->get('show_messages', 1))
{
	$active   = $unread ? ' badge-warning' : '';
	$output[] = '<div class="btn-group ' . $inboxClass . '">'
		. ($hideLinks ? '' : '<a href="' . $inboxLink . '">')
		. '<span class="badge' . $active . '">' . $unread . '</span> '
		. JText::plural('MOD_STATUS_MESSAGES', $unread)
		. ($hideLinks ? '' : '</a>')
		. '<span class="btn-group separator"></span>'
		. '</div>';
}
```
and delete the new strings.

**Comment by infograf768** [comment]
Concerning Messages, I guess the tooltip is no more necessary.

**Comment by C-Lodder** [comment]
@infograf768 - I used new language strings bacause the ones already provided add a number to the beginning of the string, hence the `%d`. The number is already shown in the `badge` so no need to show it again.

I'll double check on RTL....always forget this

**Comment by infograf768** [comment]
@C-Lodder 
Yep, saw that and deleted my post. ;)
but I still think the tooltip is now useless.

**Comment by C-Lodder** [comment]
I've now applied the change to RTL languages and also fixed the whitespace issue as mentioned by @brianteeman

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 6d6544efe2f8b57d4271ba3f21b184e2573d077b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10918">issues.joomla.org/joomla-cms/10918</a>.</sub>

**Comment by brianteeman** [comment]
Sorry to be a pain but because of some stupid rules we dont remove unused strings

**Comment by C-Lodder** [comment]
Added them back in.

Why on earth are language strings that are no longer used not removed?

**Comment by brianteeman** [comment]
Don't even go there

**Comment by MATsxm** [comment]
I have tested this item :white_check_mark: successfully on f14c5a4e7cb173e4a7e27b4ef36f46123d9f20b9<br /><br />Thanks for the great job and a big HA!!!! for the obsolete strings...<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10918">issues.joomla.org/joomla-cms/10918</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on f14c5a4e7cb173e4a7e27b4ef36f46123d9f20b9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10918">issues.joomla.org/joomla-cms/10918</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10918">issues.joomla.org/joomla-cms/10918</a>.</sub>

**Comment by mbabker** [comment]
>> Why on earth are language strings that are no longer used not removed?
>
> Don't even go there

It's about as logical as brexit.  #toosoon?

**Comment by infograf768** [comment]
Please alpha order the new lang strings. It should be:
```
MOD_STATUS_MESSAGES_0="%d Messages"
MOD_STATUS_MESSAGES_1="%d Message"
MOD_STATUS_MESSAGES_LABEL_0="Messages"
MOD_STATUS_MESSAGES_LABEL_1="Message"
MOD_STATUS_MESSAGES_LABEL_MORE="Messages"
MOD_STATUS_MESSAGES_MORE="%d Messages"
```
Also, It looks likee you did not run correctly the generatecss.php as there is a change in Hathor that was not done by a former css patch I guess. My generatecss also adds this:

```
diff --git a/administrator/templates/hathor/css/template.css b/administrator/templates/hathor/css/template.css
index 32d3ffe..d61e93c 100644
--- a/administrator/templates/hathor/css/template.css
+++ b/administrator/templates/hathor/css/template.css
@@ -3455,8 +3455,4 @@
 	margin-bottom: 10px;
 }
-fieldset.uploadform .control-group,
-fieldset.uploadform .form-actions {
-	display: inline-block;
-}
 #installer-database,
 #installer-discover,
```

**Comment by C-Lodder** [comment]
@infograf768 - alpha ordered the language strings. No idea what you mean by the `generatecss.php`. I didn't touch this at all. I simply forked the latest, made my changes, then compiled the LESS using the button in the Isis toolbar

**Comment by infograf768** [comment]
When you modify a .less file you have to run a CLI called generatecss.php, NOT use the Isis compilation.
I.e. cd to your local repository (on Mac it is called Terminal) and enter 
`php build/generatecss.php`
It will do the job for you on the .css files.

**Comment by C-Lodder** [comment]
Erm ok, never heard about this before, nor have I ever done it. Is there some sort of "Contributing guidelines" wiki that I can refer to for future reference?

**Comment by infograf768** [comment]
That's for Macintosh
http://www.macworld.co.uk/feature/mac-software/get-more-out-of-os-x-terminal-3608274/
for Linux
https://www.linux.com/learn/how-use-linux-command-line-basics-cli
I guess for windows
http://www.bleepingcomputer.com/tutorials/windows-command-prompt-introduction/

**Comment by C-Lodder** [comment]
I don't mean that. I mean a guidline as to what steps to take when contributing to Joomla. Where does it say that people need to build the `generatecss.php`?

**Comment by mbabker** [comment]
It doesn't say it explicitly anywhere but if you make a change to any LESS file in core you should run the `generatecss.php` script to ensure all resources are correctly updated.  Since your changes only affect Isis it probably isn't a major issue this time but if you changed something in the media directory as an example it would affect multiple templates.

**Comment by wilsonge** [comment]
@C-Lodder I can't merge this until you run the generateless stuff. I can run you through it if it helps?

**Comment by wilsonge** [comment]
I'm removing the RTC label for now

**Comment by C-Lodder** [comment]
I'll run it romorrow

**Comment by C-Lodder** [comment]
@wilsonge - all done

**Comment by wilsonge** [comment]
RTC

**Comment by wilsonge** [comment]
Bad bot

**Comment by zero-24** [comment]
Come on @joomla-cms-bot

**Pull request merged**